### PR TITLE
spotlight: settings based actions added

### PR DESCRIPTION
**Pull request by anwarulislam**
Closes HFE-127

### Summary
🎨🔍🌐

This pull request adds a new feature to the app: the ability to search and perform settings-related actions from the spotlight. It updates the English locale file, the `spotlight/index.vue` component, and some dependencies to support this feature. It also adds a new service class `SettingsSpotlightSearcherService` that provides the settings search logic and results.

> _We're updating the icons and the locale, me hearties_
> _We're adding a spotlight for the settings, yo ho_
> _We're heaving the code with the `pnpm` lock, me hearties_
> _We're sailing the app with the latest features, yo ho_

### Walkthrough
*  Remove unused keys and capitalize theme options in English locale file ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-31b7742ac6fc2a6ebf4cc2f4dce1ef9a78727cea803993fc23813728efe1920eL476-L477), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-31b7742ac6fc2a6ebf4cc2f4dce1ef9a78727cea803993fc23813728efe1920eL578-R579))
*  Add new keys for spotlight actions in English locale file ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-31b7742ac6fc2a6ebf4cc2f4dce1ef9a78727cea803993fc23813728efe1920eL602-R606))
*  Update version of `@iconify-json/lucide` dependency in `package.json` and `pnpm-lock.yaml` files ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-016b2ebcb9625259315d04fe283390666282f5ef4629d373957f2e2cb2073f8eL113-R113), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL641-R642), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5818-R5823))
*  Import and use `SettingsSpotlightSearcherService` in `spotlight` component ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-c55b5edadff8369ee5dfb127a95bfe09eb3757ff520bdecf2be9e24bf405482fR98), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-c55b5edadff8369ee5dfb127a95bfe09eb3757ff520bdecf2be9e24bf405482fR114))
*  Define `SettingsSpotlightSearcherService` class in `settings.searcher.ts` file ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-853eb0e692290a402b949db2329d5111829f538aab19c82be2d707b71a42a034R1-R206))
*  Update versions of `@intlify` dependencies in `pnpm-lock.yaml` file ([link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5862-R5869), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5924-R5934), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5941-R5948), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5962-R5968), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5989-R5995), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL6017-R6023), [link](https://github.com/hoppscotch/hoppscotch/pull/3244/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL6044-R6050))

### CODE REVIEW

**Review by liyasthomas — APPROVED** [review]
LGTM 💯

File: packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  alternates: string[]
+  icon: object | Component
+}
+
+/**
+ *
+ * This searcher is responsible for providing user related actions on the spotlight results.
+ *
+ * NOTE: Initializing this service registers it as a searcher with the Spotlight Service.
+ */
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts:32]: Correct the comment.

File: packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  public static readonly ID = "SETTINGS_SPOTLIGHT_SEARCHER_SERVICE"
+
+  private t = getI18n()
+
+  fontSizes = HoppFontSizes
+  activeFontSize = useSetting("FONT_SIZE")
+  isLargeFontSize = computed(
+    () =>
+      this.fontSizes.indexOf(this.activeFontSize.value) ===
+      this.fontSizes.length - 1
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts:43]:
```suggestion
      this.activeFontSize.value === "large"
```

File: packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  )
+  isSmallFontSize = computed(
+    () => this.fontSizes.indexOf(this.activeFontSize.value) === 0
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts:46]:
```suggestion
    () => this.activeFontSize.value === "small"
```

File: packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  changeTheme(theme: HoppBgColor) {
+    applySetting("BG_COLOR", theme)
+  }
+
+  changeFontSize(increase = false) {
+    const index = this.fontSizes.indexOf(this.activeFontSize.value)
+    const newIndex = index + (increase ? 1 : -1)
+    // if (newIndex < 0 || newIndex >= this.fontSizes.length) return
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts:172]: You should block/ignore the execution still or throw an error.

File: packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+  icon: object | Component
+}
+
+/**
+ *
+ * This searcher is responsible for providing user related actions on the spotlight results.
+ *
+ * NOTE: Initializing this service registers it as a searcher with the Spotlight Service.
+ */
+export class SettingsSpotlightSearcherService extends StaticSpotlightSearcherService<Doc> {
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/services/spotlight/searchers/settings.searcher.ts:33]: Specify `public` and `private` properties explicitly.

File: packages/hoppscotch-common/locales/en.json [hunk 6]
```diff
@@ -31,7 +31,6 @@
     "open_workspace": "Open workspace",
     "paste": "Paste",
     "prettify": "Prettify",
-    "rename": "Rename",
```
Comment (liyasthomas) [inline packages/hoppscotch-common/locales/en.json:34]: Why do you remove this string? Default rebase.

**Review by liyasthomas — CHANGES_REQUESTED** [review]
Why many strings from `packages/hoppscotch-common/locales/en.json` are removed?

Also why there's a change in `pnpm-lock.yaml` without any updates to `package.json`?

Can you revert these unintended changes.

**Comment by anwarulislam** [comment]
> Why many strings from `packages/hoppscotch-common/locales/en.json` are removed?
> 
> Also why there's a change in `pnpm-lock.yaml` without any updates to `package.json`?
> 
> Can you revert these unintended changes.

- Didn't remove any string from the locale file. I think it's because of rebase. I'm on it 
- I don't know about the `pnpm-lock` changes, cause I only rebased with release branch and run `pnpm i`

**Review by AndrewBastin — APPROVED** [review]

**Pull request merged**